src/main/java/seedu/address/model/person/Role.java

```diff
@@ -12,7 +12,7 @@ public enum Role {
     PROFESSOR;
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.";
+            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR', or an unambiguous prefix of it.";;
```

Comment:
There's a reason why both messages exists.

Previously if the JSON contains something like `s` instead of `STUDENT` as the role, the error message being thrown would be `Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.`. With this change, the error message would be incorrect.

Reply:
Oh that's why there was a separate error message for the role in ParseUtil. I'll fix it